cmd/oras/manifest/push.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package manifest
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"oras.land/oras-go/v2/errdef"
+	"oras.land/oras/cmd/oras/internal/display"
+	"oras.land/oras/cmd/oras/internal/option"
+	"oras.land/oras/internal/file"
+)
+
+type pushOptions struct {
+	option.Common
+	option.Descriptor
+	option.Pretty
+	option.Remote
+
+	targetRef string
+	fileRef   string
+	mediaType string
+}
+
+func pushCmd() *cobra.Command {
+	var opts pushOptions
+	cmd := &cobra.Command{
+		Use:   "push [flags] name[:tag|@digest] file",
+		Short: "[Preview] Push a manifest to remote registry",
+		Long: `[Preview] Push a manifest to remote registry
+
+** This command is in preview and under development. **
+
+Example - Push a manifest to repository 'locahost:5000/hello' and tag with 'latest':
+  oras manifest push localhost:5000/hello:latest manifest.json
+
+Example - Push a manifest with content read from stdin:
+  oras manifest push localhost:5000/hello:latest -
+
+Example - Push a manifest and output the descriptor:
+  oras manifest push --descriptor localhost:5000/hello:latest manifest.json
+
+Example - Push a manifest to repository 'locahost:5000/hello' and output the prettified descriptor:
+oras manifest push --descriptor --pretty localhost:5000/hello manifest.json
+
+Example - Push an ORAS artifact manifest to repository 'locahost:5000/hello' and tag with 'latest':
+  oras manifest push --media-type application/vnd.cncf.oras.artifact.manifest.v1+json localhost:5000/hello:latest oras_manifest.json
+
+Example - Push a manifest without TLS:
+  oras manifest push --insecure localhost:5000/hello:latest manifest.json
```

Comment:
`--insecure` is to use TLS but ignore the cert verification. Can we remove this example?